preservedata.js

```diff
@@ -0,0 +1,26 @@
+// Store data in local storage
+function storeData(data) {
+  localStorage.setItem('storedData', JSON.stringify(data));
+}
+
+// Retrieve data from local storage
+function retrieveData() {
+  return JSON.parse(localStorage.getItem('storedData')) || {};
+}
+const form = document.getElementById('form');
+const inputs = form.querySelectorAll('input');
+
+// Autofill form fields with stored data
+const storedData = retrieveData();
+inputs.forEach((input) => {
+  if (storedData[input.name]) {
+    input.value = storedData[input.name];
+  }
+});
+
+// Update stored data when input fields change
+form.addEventListener('input', (event) => {
+  const input = event.target;
+  storedData[input.name] = input.value;
+  storeData(storedData);
+});
```

Comment:
- Currently, the input added to the form doesn't save to local storage. Can you try to fix this? After filling out the form, and not submitting it, the entered values should be saved in the browser.

Here is what I can currently see 👇🏽 
<img width="682" alt="Screenshot 2023-02-03 at 18 02 51" src="https://user-images.githubusercontent.com/99246359/216732811-9408dde0-8eef-49d2-aa4c-8566834ae891.png">